lib/ansible/modules/cloud/opennebula/one_image.py

```diff
@@ -0,0 +1,519 @@
+
+    if not module.check_mode:
+        client.call('image.delete', image.id)
+        wait_for_delete(module, image)
+
+    return {'changed': True}
+
+
+def get_connection_info(module):
+
+    url = module.params.get('api_url')
+    username = module.params.get('api_username')
+    password = module.params.get('api_password')
+
+    if not url:
+        url = os.environ.get('ONE_URL')
+
+    if not username:
+        username = os.environ.get('ONE_USERNAME')
+
+    if not password:
+        password = os.environ.get('ONE_PASSWORD')
+
+    if not(url and username and password):
+        module.fail_json(msg="One or more connection parameters (api_url, api_username, api_password) were not specified")
+    from collections import namedtuple
+
+    auth_params = namedtuple('auth', ('url', 'username', 'password'))
+
+    return auth_params(url=url, username=username, password=password)
+
+
+def main():
+    fields = {
+        "api_url": {"required": False, "type": "str"},
+        "api_username": {"required": False, "type": "str"},
+        "api_password": {"required": False, "type": "str", "no_log": True},
+        "id": {"required": False, "type": "int"},
+        "name": {"required": False, "type": "str"},
+        "state": {
+            "default": "present",
+            "choices": ['present', 'absent'],
+            "type": "str"
+        },
+        "action": {
+            "choices": ['clone', 'rename', 'list'],
+            "type": "str"
+        },
+        "enabled": {"required": False, "type": "bool"},
+        "new_name": {"required": False, "type": "str"},
+        "regexp": {"required": False, "type": "str"}
+    }
+
+    module = AnsibleModule(argument_spec=fields,
+                           mutually_exclusive=[
+                               ['id', 'name', 'regexp'],
+                               ['regexp', 'state'], ['regexp', 'enabled'], ['regexp', 'new_name']
+                           ],
+                           required_if=[["action", "list", ["regexp"]]],
+                           required_one_of=[['id', 'name', 'regexp']],
+                           supports_check_mode=True)
+
+    actions = {
+        'clone': clone_image,
+        'rename': rename_image
+    }
+
+    if not HAS_OCA:
+        module.fail_json(msg='This module requires python-oca to work!')
+
+    auth = get_connection_info(module)
+    params = module.params
+    id = params.get('id')
+    name = params.get('name')
+    state = params.get('state')
+    enabled = params.get('enabled')
+    rename = params.get('rename')
+    action = params.get('action')
+    new_name = params.get('new_name')
+    regexp = params.get('regexp')
+    client = oca.Client(auth.username + ':' + auth.password, auth.url)
+
+    result = {}
+
+    if regexp and action != 'list':
+        module.fail_json(msg="Parameter 'action' has to be 'list' when the option 'regexp' is specified.")
+
+    if action and action == 'list':
```

Comment:
a list action or state is something we have accepted in the past, but not anymore for new modules. please create a separate facts module for this functionality.